### DISCUSSION TITLE
M2 and M3 Flashing blue with a solid light.

### DISCUSSION THREAD

**Discussion by noahra** [bitcraze/discussions#759] [Q&A] [answered]
![image](https://github.com/bitcraze/discussions/assets/77496609/8fb7ec48-2d65-4054-b773-e0231d76a8cd)

I have worked with my CF-drone for quite a while. Each time I plug in a battery, the rotors spin one after another as a short sound sequence chimes. But today, suddenly, that stopped happening. Instead M2 and M3 are lit up with a solid blue color and I can't connect to it with cfclient anymore. I tried disconnecting the battery etc but that didn't do anything. Anyway, then I held the power button for a couple of seconds and now M2 and M3 are instead blinking (bootloader mode). But I still cannot connect to it via cfclient: see the image:
![350314506_631540635558536_6923437392978519180_n](https://github.com/bitcraze/discussions/assets/77496609/16b6e248-195a-4a15-a326-88631cd1c6d2)

**Comment by knmcguire** [comment]
that is indeed weird! But it seems that it still can go into bootloader mode so that is good.

Could you try a coldboot recovery flashing? You can check the instructions here: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/recovery-mode/

**Reply by noahra** [reply]
It did not work. I could do "Program" but after that, the "Restart in firmware mode" button was greyed out. 

I am also realising that the battery connector of the drone is glitchy. Is it possible to buy a new one and replace the old one by soldering a new one?

**Reply by tobbeanton** [reply]
Could be a hardware problem that is preventing the Crazyflie from booting. Can you visually inspect the board and try to look for any abnormalities? Other things too try could be [DFU flashing](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/).

As for the power cable you could try shortening it, but that would only work if the cable break is close to the end.

**Reply by Guitarwang01** [reply]
I also met this problem today. This problem suddenly occured when I restared it after a normal flight. I also wonder whether it is a hardware problem.

**Reply by tobbeanton** [reply]
Sometimes it takes a long time for the system to boot if some startup tests fail. If you e.g. wait for a minute will it boot then so you can check the error?

**Reply by Guitarwang01** [reply]
I connected it to a battery for several minutes but it has no reaction. Is there any way to check what the problem is?

**Reply by tobbeanton** [reply]
No, not easily when the system doesn't boot but e.g. connecting a debugger it can be checked. Have you tried the cold-boot recovery?

**Reply by Guitarwang01** [reply]
yes, I tried to flash the latest release for several times. At the same time, I also read the discussion: https://github.com/orgs/bitcraze/discussions/172 and tried DFU update but it remains not working. 
The drone is able to connect to the client through crazyradio, and there is no heat generated while connecting to the battery, so I think probably it is no a hardware problem. But I also followed the tutorial of DFU and the responce in terminal seems normal. It is very strange.

**Reply by Guitarwang01** [reply]
By the way, i also want to describe the performance of LEDs:
M2 and M3 lid solid light when the problem occured;
after I tried cold-boot recovery, M3 was out and only M2 lid a soild light;
after I tried the DFU update, M3 was back.
At the same time, M4 performs well whenever the crazyflie connects to the crazyradio, and M1 is always out.

**Reply by tobbeanton** [reply]
You say it can connect to the cfclient. Can you post the console tab output after connecting to it? Does e.g. the propeller test in the console tab work? What is it that doesn't work? Could it be that the M1 led is broken?

**Reply by Guitarwang01** [reply]
Oh, I'm sorry that I didn't express that correctly. It cannot connect to client under flight control site, it can only connect to bootloader when the drone is under bootloader mode. The output of client stays on "connecting to radio://0/80/2M ..." and "found 0 devices". The nomal propeller test doesn't work when I turn it on. So I think maybe the drone initializing program didn't run properly.

**Comment by knmcguire** [comment]
Alright let's start again as it is a bit unclear to us what is going wrong.

- So M2 and M3 were solid when you first started. M2 and M3 being solid is normal for a crazyflie but I assume M1 was not flashing at that point. And no cfclient connection
- Cold boot didn't work. M3 was off which means that the stm flash failed. 
- With the DFU update, M3 is back

M2 stands for the NRF chip so if you don't reflash that nothing will happen there. 
M3 stands for the STM, which holds the crazyflie firmware + bootloader (that enables over air flashing), so if that is off or not responding, there is something wrong there.

So when you say that you did a DFU update, did you [update the firmware](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/#automatic-dfu-flashing) or the [bootloader](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/#bootloader-recovery). There are some very important differences between the two and if you have flashed the stm firmware over the bootloader that will break everything.

**Reply by Guitarwang01** [reply]
Thank you for your reply on the meaning of LEDs.
As for the DFU update, I updated bootloader at first and nothing changed. Then I tried update the firmware and M3 is back.
It seems that update the bootloader is a very bad operation right?

**Reply by knmcguire** [reply]
Well.. not necessary. Updating the bootloader can fix a lot of things if the crazyflie is bricked. It is normal for if you upload the bootloader nothing changes, it will just enable you to to do over air flashing with the crazyflie. But we only recommend to do it when things are really off.

So, with m2 and m3 back on, you are able to put it in bootloader mode aka push the power button for 3 seconds, and m2 and m3 both start blinking right? But flashing the latest firmware (where you can see in the loading bar underneath that it is actually flashing), as you have done multiple times does not work?

**Reply by Guitarwang01** [reply]
yes. M2 and M3 both blink right and the loading bar of flashing firmware is normal. Coldboot flash doesn't work and after flashing the latest release firmware-cf2-2023.07, M3 was out.

**Reply by tobbeanton** [reply]
> M2 and M3 both blink right and the loading bar of flashing firmware is normal.

So flashing over the radio does work? I don't see why a cold boot flash would not work then.

**Reply by Guitarwang01** [reply]
I want to make sure that does flashing over radio means the same as cold boot flash? I've tried several times and it doen't work.

**Reply by knmcguire** [reply]
Flashing over radio is a bit broad: Either a warm boot (where you are already connect to the cfclient and initate flash without pressing the button for a long time) or a cold boot (where you manually put the crazyflie in bootloader mode with a long press). Over air flashing just means that you are not using a cable or programmer to flash the crazyflie, only the crazyradio.

But sorry, my wording was different that might have confused @tobbeanton. @Guitarwang01  are able to do over air flashing with cold boot, but the firmware side of the stm is still not responding as it should upon restart. probably freezes due to an unknown reason.

**Reply by Guitarwang01** [reply, marked as answer]
There is a good news that another teammate tried flashing a previous edition flamware on his PC and it worked again! We also tried several editions and found only some of them works. The latest edition also doesn't work. That seems very strange. We are also not sure which section goes wrong duing our operation. But whatever, it works now! Thank you for your very patient explaination!

**Reply by knmcguire** [reply]
Ah no worries! and happy you found it!

But that it happened with the latest firmware is a bit worrysome.. we are just in the process of release testing.... perhaps you just had one that happened to be faulty? Could you check perhaps the commit hash (like with git log)?

**Reply by Guitarwang01** [reply]
Sorry for my late reply. We tried flash the zip from release instead of from a downloaded file, and I am not sure how to check the hash. Can you tell me something about that?

**Reply by knmcguire** [reply]
ahhh oke, no the commit hash is just if you build and flash from the github source repository. Which version of the release zip file did you use that worked out?

**Reply by Guitarwang01** [reply]
We tried many versions and I can't remember precisely. I'm sure that firmware-cf2-2023.7 doesn't work well all the time. I remember we tried this version at very first when we just start this project and it worked very well until the problem occured. We finally flashed the first version firmware-cf2-2019.02.1 and it was back again. And my teammate is sure that the versions of 2019, 2020 and 2021 works well.

**Reply by krichardsson** [reply]
OK thanks!

**Reply by Cr05512** [reply]
Hello everyone,

sorry for bringing back this topic, but I stumbled upon the same problem, which apparently still persists.
I am now working with several CF2.1, and there was this unit in particular exhibiting all of the above.
If I flash latest firmware (e.g., 2024s) it gets stuck on boot with solid blue M2 and M3. 
If, in cold boot, I flash 2020.2 or earlier, it seems to work properly.
This is limiting since many of the recent decks do not work properly with older firmwares.
Anyone found a definitive solution for this?

Best,
Alessandro

**Reply by knmcguire** [reply]
So in all situations, you have tried to flash in cold boot? Not in warmboot as the latest instructions recommend https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-upgrade

**Reply by Cr05512** [reply]
All situations. Tried cold boot flashing with different machines and cables. The only firmware that works is from 2020. Whatever later than that gives the two solid blue M2 and M3 leds :-(

**Reply by ataffanel** [reply]
Are you flashing with the latest version of the client or lib? We have updated the nRF bluetooth stack recently and only a recent client of lib is able to flash the new bluetooth stack. This could explain why only the old firmware works.

**Reply by Cr05512** [reply]
Hey Arnaud! Thanks for the reply. I have tried several machines and cables by cloning the latest client version from git. In the end, it seems I was able to get the latest firmware working, but some decks are now complaining. I start to think the unit came faulty, cause all the others I received are working pretty well. That cf is the only one complaining 😄

**Reply by ataffanel** [reply]
What are the decks complaining about? What error message are you getting?

But yes, if this is the only cf that has problems it is possible there is something wrong about it.  If we cannot find anything software or configuration related please drop a mail at support_at_bitcraze.io and we should be able to fix a replacement

**Reply by Cr05512** [reply]
Sorry for the sparse responses, been very absorbed lately. In the end I realised that the deck complaining was... faulty. Multiranger had a "not OK" initialization on one of the ToF sensors. I more or less could get the CF2.1 working again (required several flashing attempts). Nonetheless, I don't know yet (apart from faulty decks) what was the reason behind failed flashings :-(

I have few units still left for testing, I will double check if the problem was isolated or could be batch-related. 

In the meanwhile, thanks for the support.
